src/main/scala/com/github/traviscrawford/spark/dynamodb/BaseScanner.scala

```diff
@@ -99,10 +94,66 @@ private[dynamodb] trait BaseScanner {
 
     scanSpec
   }
+
+  def getQuerySpec(config: Config): QuerySpec = {
+    val querySpec = new QuerySpec()
+      .withMaxPageSize(config.pageSize)
+      .withReturnConsumedCapacity(ReturnConsumedCapacity.TOTAL)
+
+    // Queries might have required partition/hash keys
+    val partitionKey = parseStringToMap(config.maybePartitionKey)
+    val sortKey = parseStringToMap(config.maybeSortKey)
+
+    // Parse any projection expressions passed in
+    val exprSpecBuilder = config.maybeRequiredColumns
+      .map(requiredColumns => new ExpressionSpecBuilder().addProjections(requiredColumns: _*))
+    val parsedProjectionExpr = exprSpecBuilder.map(xSpecBuilder => xSpecBuilder.buildForScan())
+    val projectionNameMap = parsedProjectionExpr.flatMap(projExpr => Option(projExpr.getNameMap))
+      .map(_.asScala.toMap).getOrElse(Map.empty)
+    val projectionValueMap = parsedProjectionExpr.flatMap(projExpr => Option(projExpr.getValueMap))
```

Comment:
@Chuwiey a simple Scala question:   foo.flatMap(x => Option(x))  is same as foo.map(x => x), basically is same as foo?  I must be missing something